### PR TITLE
feat(cdk/table): add event that emits after rendering

### LINKED ISSUE #8068: [Table] Emit event when after rendering rows
#### Bug, feature request, or proposal:
Resquest

#### What is the expected behavior?
Have an event that is emite when the table is rendered

#### What is the current behavior?
I have an example of a table with a lot of data and while not painting the table is blank and I would like to show a loading while it does not finish rendering the table.
i only can control the call to service but not the rendered of the table.
 is there any way to know when I finish rendering the table?

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
"peerDependencies": {
    "@angular/flex-layout": "2.0.0-beta.9",
    "@angular/material": "2.0.0-beta.10",
    "@angular/cdk": "2.0.0-beta.10",
    "@angular/animations": "4.3.6",
    "@angular/common": "4.3.6",
    "@angular/compiler": "4.3.6",
    "@angular/compiler-cli": "4.3.6",
    "@angular/core": "4.3.6",
    "@angular/forms": "4.3.6",
    "@angular/http": "4.3.6",
    "@angular/platform-browser": "4.3.6",
    "@angular/platform-browser-dynamic": "4.3.6",
    "@angular/platform-server": "4.3.6",
    "@angular/router": "4.3.6",
    "@angular/upgrade": "4.3.6",
    "@angular/cli": "1.3.2",
    "@ngx-translate/core": "8.0.0",
    "@ngx-translate/http-loader": "1.0.2",
    "core-js": "^2.4.1",
    "rxjs": "^5.4.2",
    "zone.js": "^0.8.12",
    "reflect-metadata": "0.1.10"
  },

### PR DESCRIPTION
**Pull request by andrewseguin**
It's not uncommon for a client to want to respond to the table after it finishes rendering rows. You could subscribe to the data source and watch for changes there, but there's no good timing to know that the DOM is actually changed.

Fixes #8068

### CODE REVIEW

File: src/cdk/table/table.ts [hunk 1]
```diff
@@ -612,6 +621,7 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     const changes = this._dataDiffer.diff(this._renderRows);
     if (!changes) {
       this._updateNoDataRow();
+      this.dataRowsChange.next();
```
Comment (crisbeto) [inline src/cdk/table/table.ts:624]: Doesn't this call mean that it'll fire the event even if the rows haven't changed?

File: src/cdk/table/table.ts [hunk 2]
```diff
@@ -455,6 +457,13 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
   }
   private _fixedLayout: boolean = false;
 
+  /**
+   * Emits when the table completes rendering a set of data rows based on the latest data from the
+   * data source, even if the set of rows is empty.
+   */
+  @Output()
+  readonly dataRowsChange = new EventEmitter<void>();
```
Comment (crisbeto) [inline src/cdk/table/table.ts:465]: I'm not 100% sure on the naming. Wouldn't it also fire if the rows remained the same, but some of cells changed?

Comment (andrewseguin) [reply src/cdk/table/table.ts:465]: Good point, it's not quite representing that the rows changed, just that we attempted a render based on a data change. I think the placement of the `emit` is what clients want (they just want to know the table reacted to the data). Any suggestions on the name?

Comment (crisbeto) [reply src/cdk/table/table.ts:465]: What about something like `dataChanged` or `contentChanged`?

Comment (andrewseguin) [reply src/cdk/table/table.ts]: Renamed to `contentChanged`, SGTM

**Comment by pccosta-pdm** [comment]
Hey guys, any update on this issue? I would be very happy to have this applied for use.

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**